src/rpc/rawtransaction.cpp

```diff
@@ -1773,6 +1774,324 @@ UniValue converttopsbt(const JSONRPCRequest& request)
+    }
+
+    for (unsigned int i = 0; i < txs.size(); ++i) {
+        PartiallySignedTransaction psbtx;
+        std::string error;
+        if (!DecodePSBT(psbtx, txs[i].get_str(), error)) {
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR, strprintf("TX decode failed %s", error));
+        }
+        psbtxs.push_back(psbtx);
+    }
+
+    PartiallySignedTransaction merged_psbt(psbtxs[0]); // Copy the first one
+
+    // Clear signatures from all inputs
+    for (auto& input : merged_psbt.inputs) {
+        input.partial_sigs.clear();
+    }
+
+    // Merge
+    for (auto it = std::next(psbtxs.begin()); it != psbtxs.end(); ++it) {
+        for (unsigned int i = 0; i < it->tx->vin.size(); ++i) {
+            if (!merged_psbt.AddInput(it->tx->vin[i], it->inputs[i])) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "An input exists in multiple PSBTs");
+            }
+        }
+        for (unsigned int i = 0; i < it->tx->vout.size(); ++i) {
+            merged_psbt.AddOutput(it->tx->vout[i], it->outputs[i]);
+        }
+    }
+
+    CDataStream ssTx(SER_NETWORK, PROTOCOL_VERSION);
+    ssTx << merged_psbt;
+    return EncodeBase64((unsigned char*)ssTx.data(), ssTx.size());
+}
+
+UniValue analyzepsbt(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+            "analyzepsbt \"psbt\"\n"
+            "\nAnalyzes and provides information about the current status of a PSBT and its inputs\n"
+            "\nArguments:\n"
+            "1. \"psbt\"             (string) A base64 string of a PSBT\n"
+
+            "\nResult:\n"
+            "{\n"
+            "  \"inputs\" : [                      (array of json objects)\n"
+            "    {\n"
+            "      \"has_utxo\" : true|false     (boolean) Whether a UTXO is provided\n"
+            "      \"is_final\" : true|false     (boolean) Whether the input is finalized\n"
+            "      \"missing\" : {               (json object, optional) Things that are missing that are required to complete this input\n"
+            "        \"pubkeys\" : [             (array)\n"
+            "          \"keyid\"                 (string) Public key ID, hash160 of the public key, of a public key whose BIP 32 derivation path is missing\n"
+            "        ]\n"
+            "        \"signatures\" : [          (array)\n"
+            "          \"keyid\"                 (string) Public key ID, hash160 of the public key, of a public key whose signature is missing\n"
+            "        ]\n"
+            "        \"redeemscript\" : \"hash\"   (string) Hash160 of the redeemScript that is missing\n"
+            "        \"witnessscript\" : \"hash\"  (string) SHA256 of the witnessScript that is missing\n"
+            "        \"next\" : \"role\"           (string) Role of the next person that this input needs to go to\n"
```

Comment:
This doesn't match the code, right? "next" goes inside the input, not inside "missing"?